docs/docs/aleph_zero/sdk.md

```diff
@@ -0,0 +1,386 @@
+:::
+
+:::info Creating types
+You can create custom decimals with `toDecimal` syntax. First argument is an integer and second argument is a number of zeros. For example `toFee(3n, 2n)` will produce decimal equal to `0.03`. Read more about types here: [Types](types.md).
+:::
+
+:::warning Token sorting
+Tokens are sorted alphabetically when pool key is created, so make sure that you swap tokens in correct direction. Read more about pool keys here: [PoolKey](storage#poolkey).
+:::
+
+```typescript
+const main = async () => {
+  // --snip--
+
+  // load invariant contract
+  const invariant = await Invariant.load(api, Network.Main, INVARIANT_ADDRESS)
+
+  // load token contracts
+  const token0 = await PSP22.load(api, Network.Main, TOKEN0_ADDRESS) // token we want to give
+  const token1 = await PSP22.load(api, Network.Main, TOKEN1_ADDRESS) // token we want to receive
+
+  // set fee tier and pool key, make sure that pool with specified parameters exists
+  const feeTier = newFeeTier(toFee(1n, 2n), 1n) // fee: 0.01 = 1%, tick spacing: 1
+  const poolKey = newPoolKey(
+    token0.contract.address.toString(),
+    token1.contract.address.toString(),
+    feeTier
+  )
+
+  // make sure that you swap in correct direction,
+  // here we check if token0 (token we want to give) is tokenX
+  let xToY = isTokenX(token0.contract.address.toString(), token1.contract.address.toString())
+}
+
+main()
+```
+
+### Perform swap
+
+:::info How to calculate input amount
+In order to calculate input amount, we have to multiply actual token amount you want to swap times 10 to the power of decimal.
+Let's say some token has decimal of 12 and we want to swap 6 actual tokens. Here is how we can calculate input amount:
+6 \* 10^12 = 6000000000000.
+:::
+
+```typescript
+const main = async () => {
+  // --snip--
+
+  // here we want to swap 6 token0
+  // token0 has 12 decimals so we need to multiply it by 10^12
+  const amount = 6n * 10n ** 12n
+
+  // approve token0 transfer
+  await token0.approve(account, invariant.contract.address.toString(), amount)
+
+  // slippage is a price change you are willing to accept,
+  // for examples if current price is 1 and your slippage is 1%, then price limit will be 1.01
+  const allowedSlippage = toPercentage(1n, 3n) // 0.001 = 0.1%
+
+  // swap message only accepts price limit so we have to calculate it in order to use slippage,
+  // here we are passing current price and allowed slippage to calculate price limit
+  const pool = await invariant.getPool(
+    account,
+    token0.contract.address.toString(),
+    token1.contract.address.toString(),
+    feeTier
+  )
+  const sqrtPriceLimit = calculateSqrtPriceAfterSlippage(pool.sqrtPrice, allowedSlippage, xToY)
+
+  const result = await invariant.swap(account, poolKey, xToY, amount, true, priceLimit)
+  console.log(result.hash) // print transaction hash
```

Comment:
If you add `console.log` statement to the code snippet, it will be helpful for the reader to display the results of these logs below. I've seen this pattern many times in other docs